src/AzureAppConfigurationImpl.ts

```diff
@@ -93,35 +94,38 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
         }
 
         if (options?.refreshOptions?.enabled) {
-            const { watchedSettings, refreshIntervalInMs } = options.refreshOptions;
-            // validate watched settings
-            if (watchedSettings === undefined || watchedSettings.length === 0) {
-                throw new Error("Refresh is enabled but no watched settings are specified.");
+            const { watchedSettings, refreshIntervalInMs, watchAll } = options.refreshOptions;
+            // validate refresh options
+            if (watchAll !== true) {
+                if (watchedSettings === undefined || watchedSettings.length === 0) {
+                    throw new Error("Refresh is enabled but no watched settings are specified.");

```

Comment:
It's expected to watchAll if user enabled refresh but not specified sentinel keys